### PR TITLE
plasma-infra(plasma-new-hope): Migrate to storybook 7 [Part 4]

### PR DESCRIPTION
**Pull request by Yakutoc**
## Release Notes

Переход на `Storybook@latest(7.5.3)`:

### What/why Changed

[From version 6.5.x to 7.0.0](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#from-version-65x-to-700)

- [x] rewrite `.stories` to CSF 3
- [x] [rewrite `main.ts`](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#esm-format-in-mainjs)
- [x] [rewrite `preview.ts` ](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#default-export-in-previewjs)

<h3>⚡ Component performance testing</h3>

**Result:** 🟢 OK

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.242.1-canary.840.6823033008.0
  npm install @salutejs/plasma-colors@0.10.1-canary.840.6823033008.0
  npm install @salutejs/plasma-core@1.137.1-canary.840.6823033008.0
  npm install @salutejs/plasma-hope@1.240.1-canary.840.6823033008.0
  npm install @salutejs/plasma-icons@1.169.1-canary.840.6823033008.0
  npm install @salutejs/plasma-new-hope@0.10.1-canary.840.6823033008.0
  npm install @salutejs/cra-template-plasma-shop-template@2.4.1-canary.840.6823033008.0
  npm install @salutejs/plasma-temple@1.187.1-canary.840.6823033008.0
  npm install @salutejs/plasma-tokens-b2b@1.29.1-canary.840.6823033008.0
  npm install @salutejs/plasma-tokens-b2c@0.38.1-canary.840.6823033008.0
  npm install @salutejs/plasma-tokens-core@0.4.1-canary.840.6823033008.0
  npm install @salutejs/plasma-tokens-web@1.44.1-canary.840.6823033008.0
  npm install @salutejs/plasma-tokens@1.64.1-canary.840.6823033008.0
  npm install @salutejs/plasma-typo@0.38.1-canary.840.6823033008.0
  npm install @salutejs/plasma-ui@1.219.1-canary.840.6823033008.0
  npm install @salutejs/plasma-web@1.242.1-canary.840.6823033008.0
  npm install @salutejs/plasma-cy-utils@0.71.1-canary.840.6823033008.0
  npm install @salutejs/plasma-sb-utils@0.135.1-canary.840.6823033008.0
  npm install @salutejs/plasma-tokens-utils@0.33.1-canary.840.6823033008.0
  # or 
  yarn add @salutejs/plasma-b2c@1.242.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-colors@0.10.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-core@1.137.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-hope@1.240.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-icons@1.169.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-new-hope@0.10.1-canary.840.6823033008.0
  yarn add @salutejs/cra-template-plasma-shop-template@2.4.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-temple@1.187.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-tokens-b2b@1.29.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-tokens-b2c@0.38.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-tokens-core@0.4.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-tokens-web@1.44.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-tokens@1.64.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-typo@0.38.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-ui@1.219.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-web@1.242.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-cy-utils@0.71.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-sb-utils@0.135.1-canary.840.6823033008.0
  yarn add @salutejs/plasma-tokens-utils@0.33.1-canary.840.6823033008.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-840/

File: packages/plasma-new-hope/src/examples/_helpers.tsx [hunk 1]
```diff
@@ -32,21 +32,28 @@ export function argTypesFromConfig<
...
+
     for (const [key, val] of filteredVariations) {
         const control: ArgTypes[''] = {
             control: {},
         };
+
         if (val.true) {
             control.control.type = 'boolean';
-            control.defaultValue = false;
+            control.table = { defaultValue: { summary: false } };
```
Comment (Yakutoc) [inline packages/plasma-new-hope/src/examples/_helpers.tsx:43]: [Note]:

https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#no-longer-inferring-default-values-of-args

File: packages/plasma-web/package.json [hunk 2]
```diff
@@ -64,7 +64,7 @@
         "react": "18.2.0",
         "react-dom": "18.2.0",
         "styled-components": "5.3.1",
-        "typescript": "4.0.8",
+        "typescript": "4.2.4",
```
Comment (Yakutoc) [inline packages/plasma-web/package.json:67]: [Note]: 

Иначе будет ошибка при сборке types. так как версия TS в new-hope тоже 4.2.4

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-840/

File: packages/plasma-b2c/package.json [hunk 3]
```diff
@@ -86,7 +86,7 @@
         "react-dom": "18.2.0",
         "styled-components": "5.3.1",
         "ts-node": "10.3.0",
-        "typescript": "4.0.8",
+        "typescript": "4.2.4",
```
Comment (Yakutoc) [inline packages/plasma-b2c/package.json:89]: [Note]: 

Иначе будет ошибка при сборке types. так как версия TS в new-hope тоже 4.2.4

File: packages/plasma-new-hope/package.json [hunk 4]
```diff
@@ -88,7 +88,7 @@
         "react-dom": "18.2.0",
         "rollup": "^3.28.0",
         "styled-components": "^5.3.1",
-        "typescript": "4.0.8",
+        "typescript": "4.2.4",
```
Comment (Yakutoc) [inline packages/plasma-new-hope/package.json:92]: [Note]: 

Чтобы не устанавливать кучу пакетов аля types@package проще сделать минимальный bump версии TS. 

Ошибок сборки не было обнаружено.

File: packages/plasma-new-hope/.storybook/main.ts [hunk 5]
```diff
@@ -1,32 +1,43 @@
...
             },
             build: {
-                ...config.build,
                 sourcemap: false,
             },
             plugins: [
-                ...config.plugins,
                 linaria({
                     exclude: ['../../../'],
+                    babelOptions: {
```
Comment (Yakutoc) [inline packages/plasma-new-hope/.storybook/main.ts:34]: https://github.com/callstack/linaria/blob/master/docs/BUNDLERS_INTEGRATION.md#Vite

> If you are using language features that requires a babel transform (such as typescript), ensure the proper babel presets or plugins are passed to linaria.

```ts
import { defineConfig } from 'vite';
import linaria from '@linaria/vite';

// example to support typescript syntax:
export default defineConfig(() => ({
  // ...
  plugins: [
    linaria({
      include: ['**/*.{ts,tsx}'],
      babelOptions: {
        presets: ['@babel/preset-typescript', '@babel/preset-react'],
      },
    }),
  ],
}));
```

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-840/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-840/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-840/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-840/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-840/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-840/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-840/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-840/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-840/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-840/
b2c storybook: http://plasma.sberdevices.ru/pr/pr-840/b2c-storybook/
web storybook: http://plasma.sberdevices.ru/pr/pr-840/web-storybook/
ui storybook: http://plasma.sberdevices.ru/pr/pr-840/ui-storybook/
temple storybook: http://plasma.sberdevices.ru/pr/pr-840/temple-storybook/
new-hope storybook: http://plasma.sberdevices.ru/pr/pr-840/new-hope-storybook/
asdk storybook: http://plasma.sberdevices.ru/pr/pr-840/asdk-storybook/

**Pull request closed without merging**